### DISCUSSION TITLE
Annual Oracle data prep

### DISCUSSION THREAD

**Discussion by Jason-Conner-NOAA** [afsc-gap-products/model-based-indices#44] [Team Posts]
An important note for all Modsquadders:

Each year, after the survey data are finalized, this Oracle procedure must be run:
https://github.com/coleary-noaa/model-based-indices/blob/main/SQL/update_survey.sql

You should be able to do this using sumfish::updateSurvey()
https://github.com/afsc-gap-products/sumfish/blob/main/R/updateSurvey.R

This code syncs cruisejoins and valid VAST hauls found in the SAFE schema, tables:
SURVEY
EBS_VAST_HAULS
NBS_VAST_HAULS

This **MUST** be done prior to executing the data_prep R script.

**Comment by CaitlinAkselrud-NOAA** [comment]
Does this need to be done just once, or does each individual pulling Oracle data need to run this script?

**Comment by Jason-Conner-NOAA** [comment]
Just once conceptually and only one person needs to do it, however if something needs to be run before all the data are finalized (e.g. the EBS needs to run before the NBS data are done), then it should be executed before each case.
